packages/contracts-bedrock/test/L2/CrossL2Inbox.t.sol

```diff
@@ -0,0 +1,154 @@
+    function setUp() public {
+        crossL2Inbox = new CrossL2Inbox();
+    }
+
+    /// @dev Tests that `executeMessage` succeeds when called with valid parameters.
+    function testFuzz_executeMessage_succeeds(
+        bytes calldata _msg,
+        ICrossL2Inbox.Identifier calldata _id,
+        address _target,
+        uint256 _value
+    )
+        external
+        payable
+    {
+        vm.assume(_id.timestamp <= block.timestamp);
+        vm.assume(_target.code.length == 0);
+
+        // need to prevent call to L1Block.isInDependencySet from reverting
+        vm.mockCall({
+            callee: Predeploys.L1_BLOCK_ATTRIBUTES,
+            data: abi.encodeWithSelector(L1Block.isInDependencySet.selector, _id.chainId),
+            returnData: abi.encode(true)
+        });
+
+        vm.deal(tx.origin, _value);
+
+        // executeMessage
+        vm.prank(tx.origin);
+        vm.expectCall(_target, _value, _msg);
+        crossL2Inbox.executeMessage{ value: _value }({ _id: _id, _target: _target, _msg: _msg });
+
+        assertEq(crossL2Inbox.origin(), _id.origin);
+        assertEq(crossL2Inbox.blocknumber(), _id.blocknumber);
+        assertEq(crossL2Inbox.logIndex(), _id.logIndex);
+        assertEq(crossL2Inbox.timestamp(), _id.timestamp);
+        assertEq(crossL2Inbox.chainId(), _id.chainId);
+    }
+
+    /// @dev Tests that `executeMessage` fails when called with an identifier with an invalid timestamp.
+    function test_executeMessage_invalidTimestamp_fails() external {
+        sampleIdentifier.timestamp = block.timestamp + 1;
+
+        vm.prank(tx.origin);
+        vm.expectRevert("CrossL2Inbox: invalid id timestamp");
+        crossL2Inbox.executeMessage({ _id: sampleIdentifier, _target: address(0), _msg: hex"1234" });
+    }
+
+    /// @dev Tests that `executeMessage` fails when called with an identifier with an invalid chain ID.
+    function test_executeMessage_invalidChainId_fails() external {
+        vm.mockCall({
+            callee: Predeploys.L1_BLOCK_ATTRIBUTES,
+            data: abi.encodeWithSelector(L1Block.isInDependencySet.selector, sampleIdentifier.chainId),
+            returnData: abi.encode(false)
+        });
+
+        vm.prank(tx.origin);
+        vm.expectRevert("CrossL2Inbox: id chain not in dependency set");
+        crossL2Inbox.executeMessage({ _id: sampleIdentifier, _target: address(0), _msg: hex"1234" });
+    }
+
+    /// @dev Tests that `executeMessage` succeeds when called with an identifier with the same chain ID as
+    ///      the current chain.
+    function test_executeMessage_sameChainId_succeeds() external {
+        vm.mockCall({
+            callee: Predeploys.L1_BLOCK_ATTRIBUTES,
+            data: abi.encodeWithSelector(L1Block.isInDependencySet.selector, sampleIdentifier.chainId),
+            returnData: abi.encode(true)
+        });
+
+        vm.prank(tx.origin);
+        crossL2Inbox.executeMessage({ _id: sampleIdentifier, _target: address(0), _msg: hex"1234" });
```

Comment:
Adding a `vm.expectCall` here would make this test stronger, it is testing that there is no revert, if you are aware of the implementation details then you would know that the call is made if no revert, its good to write tests assuming a black box of the implementation